scripts/add_check.py

```diff
@@ -0,0 +1,136 @@
+        content = fp.read()
+
+    with open(os.path.join(ROOT_DIR, f"src/{dir_name(plugin)}/mod.rs"), "w") as fp:
+        for line in content.splitlines():
+            if line.strip() == "fn checks(check_code: CheckCode, path: &Path) -> Result<()> {":
+                indent = line.split("fn checks(check_code: CheckCode, path: &Path) -> Result<()> {")[0]
+                fp.write(f'{indent}#[test_case(CheckCode::{code}, Path::new("{code}.py"); "{code}")]')
+                fp.write("\n")
+
+            fp.write(line)
+            fp.write("\n")
+
+    # Add the relevant plugin function.
+    with open(os.path.join(ROOT_DIR, f"src/{dir_name(plugin)}/plugins.rs"), "a") as fp:
+        fp.write(
+            f"""
+/// {code}
+pub fn {snake_case(name)}(checker: &mut Checker) {{}}
+"""
+        )
+        fp.write("\n")
+
+    # Add the relevant sections to `src/registry.rs`.
+    with open(os.path.join(ROOT_DIR, "src/registry.rs"), "r") as fp:
+        content = fp.read()
+
+    index = 0
+    with open(os.path.join(ROOT_DIR, "src/registry.rs"), "w") as fp:
+        for line in content.splitlines():
+            fp.write(line)
+            fp.write("\n")
+
+            if line.strip() == f"// {plugin}":
+                if index == 0:
+                    # `CheckCode` definition
+                    indent = line.split(f"// {plugin}")[0]
+                    fp.write(f"{indent}{code},")
+                    fp.write("\n")
+
+                elif index == 1:
+                    # `CheckKind` definition
+                    indent = line.split(f"// {plugin}")[0]
+                    fp.write(f"{indent}{name},")
+                    fp.write("\n")
+
+                elif index == 2:
+                    # `CheckCode#kind()`
+                    indent = line.split(f"// {plugin}")[0]
+                    fp.write(f"{indent}CheckCode::{code} => CheckKind::{name},")
+                    fp.write("\n")
+
+                elif index == 3:
+                    # `CheckCode#category()`
+                    indent = line.split(f"// {plugin}")[0]
+                    fp.write(f"{indent}CheckCode::{code} => CheckCategory::{pascal_case(plugin)},")
+                    fp.write("\n")
+
+                elif index == 4:
+                    # `CheckKind#code()`
+                    indent = line.split(f"// {plugin}")[0]
+                    fp.write(f"{indent}CheckKind::{name} => &CheckCode::{code},")
+                    fp.write("\n")
+
+                elif index == 5:
+                    # `CheckCode#body`
+                    indent = line.split(f"// {plugin}")[0]
+                    fp.write(f'{indent}CheckKind::{name} => "".to_string(),')
+                    fp.write("\n")
+
+                index += 1
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description="Generate boilerplate for a new check.",
+        epilog=("python scripts/add_check.py --name PreferListBuiltin --code PIE807 --plugin flake8-pie"),
```

Comment:
Since these arguments are required, I think I'd rather make them positional e.g.

```
python scripts/add_check.py flake8-pie PIE807 PreferListBuiltin
```

Reply:
My gripe with that is that you then have to remember the order.

Reply:
Right, I think that could be mitigated by doing some input validation and providing nice error messages e.g:

* providing a nice error message if the first argument isn't an existing plugin
* providing a nice error message if the second argument doesn't have the prefix of the plugin